client/checkout/utils/platform-checkout.js

```diff
@@ -6,13 +13,69 @@ export const handlePlatformCheckoutEmailInput = ( field ) => {
 	const parentDiv = platformCheckoutEmailInput.parentNode;
 	spinner.classList.add( 'wc-block-components-spinner' );
 
-	const platformCheckoutLocateUser = () => {
+	const iframeWrapper = document.createElement( 'div' );
+	iframeWrapper.setAttribute( 'role', 'dialog' );
+	iframeWrapper.setAttribute( 'aria-modal', 'true' );
+	iframeWrapper.classList.add( 'platform-checkout-sms-otp-iframe-wrapper' );
+	const iframe = document.createElement( 'iframe' );
+	iframe.title = __(
+		'Platform checkout SMS code verification',
+		'woocommerce-payments'
+	);
+	iframe.classList.add( 'platform-checkout-sms-otp-iframe' );
+	iframe.addEventListener( 'load', () => {
+		iframe.classList.add( 'open' );
+	} );
+	iframeWrapper.insertBefore( iframe, null );
+
+	const closeIframe = () => {
+		iframeWrapper.remove();
+		iframe.classList.remove( 'open' );
+		platformCheckoutEmailInput.focus();
+	};
+	iframeWrapper.addEventListener( 'click', closeIframe );
+	window.addEventListener( 'message', ( e ) => {
+		if ( getConfig( 'platformCheckoutHost' ) !== e.origin ) {
+			return;
+		}
+
+		if ( 'close_modal' === e.data.action ) {
+			closeIframe();
+		}
+	} );
+
+	const openIframe = ( email ) => {
+		iframe.src =
+			getConfig( 'platformCheckoutHost' ) + '/sms-otp/?email=' + email;
+		parentDiv.insertBefore( iframeWrapper, null );
+		iframe.focus();
+	};
+
+	document.addEventListener( 'keyup', ( event ) => {
+		const key = event.which || event.keyCode;
+
+		if ( 27 === key && closeIframe() ) {
+			event.stopPropagation();
+		}
+	} );
+
+	const platformCheckoutLocateUser = ( email ) => {
 		parentDiv.insertBefore( spinner, platformCheckoutEmailInput );
 
-		// Placeholder to simulate request. Replace with request to Platform Checkout email verification endpoint.
-		setTimeout( () => {
-			spinner.remove();
-		}, 3000 );
+		fetch(
+			getConfig( 'platformCheckoutHost' ) +
+				'/wp-json/platform-checkout/v1/user/exists?email=' +
+				email
+		)
+			.then( ( response ) => response.json() )
+			.then( ( data ) => {
+				if ( data[ 'user-exists' ] ) {
+					openIframe( email );
+				}
+			} )
+			.finally( () => {
+				spinner.remove();
+			} );
 	};
 
 	const validateEmail = ( value ) => {
```

Comment:
I know it's not part of this PR, but the `input.checkValidity()` is not working properly for me. It's returning `true` for any value containing at least one character, followed by a `@`,  followed by at least one character. So the value `a@a` returns `true`. This is causing the `platformCheckoutLocateUser()` function to be called without a proper email address which results in a `400 Bad Request`. Ideally, we'd validate the email address before making that request.

FWIW, it's not working [outside of the codebase](https://jsfiddle.net/e9um57d0/) either.

Reply:
`a@a` is technically a valid email address that could be used by some systems so the browser html5 based validation works. Ideally we'd use the same validation scheme that is used by WordPress but this is more complicated. I could probably use a simple regex.

Reply:
> a@a is technically a valid email address that could be used by some systems
Ah, that's right. As you said, we should probably use the same validation as WordPress since all of our users will have to have an email address that WordPress sees as valid.